src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,81 @@
-import { useState } from 'react';
+import React, { useState, FormEvent } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({onAdd}) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    setCount(C => C + 1);
+
+    onAdd({ title, description, imdbUrl, imdbId, imgUrl });
+    
+    setTitle('');
+    setDescription('');
+    setImdbUrl('');
+    setImdbId('');
+    setImgUrl('');
+  };
+  const isDisabled = !title || !imdbUrl || !imdbId || !imgUrl;
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(movieTitle) => setTitle(movieTitle)}
```

Comment:
try to rewrite in all TextFields this way 👇
```suggestion
        onChange={setTitle}
```